x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.ts

```diff
@@ -0,0 +1,156 @@
+  onUserRefresh(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+}
+
+export const ApiLogsLogic = kea<MakeLogicType<ApiLogsValues, ApiLogsActions>>({
+  path: ['enterprise_search', 'app_search', 'api_logs_logic'],
+  actions: () => ({
+    fetchApiLogs: ({ isPoll } = { isPoll: false }) => ({ isPoll }),
+    pollForApiLogs: true,
+    onPollStart: (intervalId) => ({ intervalId }),
+    onPollInterval: ({ results, meta }) => ({ results, meta }),
+    storePolledData: ({ results, meta }) => ({ results, meta }),
+    updateView: ({ results, meta }) => ({ results, meta }),
+    onUserRefresh: true,
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        updateView: () => false,
+        onPaginate: () => true,
+      },
+    ],
+    apiLogs: [
+      [],
+      {
+        updateView: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        updateView: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => updateMetaPageIndex(state, newPageIndex),
+      },
+    ],
+    hasNewData: [
+      false,
+      {
+        storePolledData: () => true,
+        updateView: () => false,
+      },
+    ],
+    polledData: [
+      {} as ApiLogsData,
+      {
+        storePolledData: (_, data) => data,
+      },
+    ],
+    intervalId: [
+      null,
+      {
+        onPollStart: (_, { intervalId }) => intervalId,
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    pollForApiLogs: () => {
+      if (values.intervalId) return; // Ensure we only have one poll at a time
+
+      const id = window.setInterval(() => actions.fetchApiLogs({ isPoll: true }), POLLING_DURATION);
+      actions.onPollStart(id);
+    },
+    fetchApiLogs: async ({ isPoll }) => {
+      if (isPoll && values.dataLoading) return; // Manual fetches (i.e. user pagination) should override polling
+
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(`/api/app_search/engines/${engineName}/api_logs`, {
+          query: {
+            'page[current]': values.meta.page.current,
+            'filters[date][from]': getDateString(-1),
+            'filters[date][to]': getDateString(),
+            sort_direction: 'desc',
+          },
+        });
+
+        // Manual fetches (e.g. page load, user pagination) should update the view immediately,
+        // while polls are stored in-state until the user manually triggers the 'Refresh' action
+        if (isPoll) {
+          actions.onPollInterval(response);
+        } else {
+          actions.updateView(response);
+        }
+      } catch (e) {
+        if (isPoll) {
+          // If polling fails, it will typically be due due to http connection -
+          // we should send a more human-readable message if so
+          setErrorMessage(POLLING_ERROR_MESSAGE);
```

Comment:
Thanks for this suggestion during sync today y'all - I'm also conferring with Davey now to see if I can switch to toasts for these, that way we don't have to worry about clearing flash messages if a user's HTTP connection is interrupted and then returns later on